src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2ApplicationTest.kt

```diff
@@ -309,22 +312,22 @@ class Cas2ApplicationTest : IntegrationTestBase() {
     }
 
     @Test
-    fun `Get list of applications returns 500 when a person cannot be found`() {
+    fun `Get list of applications returns successfully when a person cannot be found`() {
       `Given a CAS2 User`() { userEntity, jwt ->
         val crn = "X1234"
 
         produceAndPersistBasicApplication(crn, userEntity)
-        CommunityAPI_mockNotFoundOffenderDetailsCall(crn)
-        loadPreemptiveCacheForOffenderDetails(crn)
+        val mockOffenderService = mockk<OffenderService>()
+        // val mockOffenderDetailsDataSource = mockk<OffenderDetailsDataSource>()
+        // every {mockOffenderDetailsDataSource.getOffenderDetailSummaries(any())} returns emptyMap()
+        every { mockOffenderService.getOffenderSummariesByCrns(any()) } returns emptyMap()
 
         webTestClient.get()
           .uri("/cas2/applications")
           .header("Authorization", "Bearer $jwt")
           .exchange()
           .expectStatus()
-          .is5xxServerError
-          .expectBody()
-          .jsonPath("$.detail").isEqualTo("Unable to get Person via crn: $crn")
+          .isOk
```

Comment:
This test is failing due to a `ClassCastException` but I'm not sure why since we should be setting personInfo to an emptyMap() and getting an object from a map using a non-existing key should return null.

Reply:
Also would be good to talk over the difference between this test and `Get list of applications returns successfully when the person cannot be fetched from the prisons API()`.

Reply:
> Also would be good to talk over the difference between this test and Get list of applications returns successfully when the person cannot be fetched from the prisons API().

The test for 'when the person cannot be fetched from the prisons API' is regarding the `inmateDetails` we get from the `PrisonApi` whereas we get the `offenderDetails` from the `OffenderDetailsDataSource` / aka AP_DELIUS_CONTEXT...
you can see in the OffenderService when we make a call to get the inmate details, if it wasn't a success we just set the details to null
```
    val inmateDetails = offender.otherIds.nomsNumber?.let { nomsNumber ->
      when (val inmateDetailsResult = getInmateDetailByNomsNumber(offender.otherIds.crn, nomsNumber)) {
        is AuthorisableActionResult.Success -> inmateDetailsResult.entity
        else -> null
      }
    }
 ```